### PR TITLE
cookie-cutter-redis: add support for multiple streams

### PR DESCRIPTION
**Pull request by mag415**

### CODE REVIEW

File: packages/redis/src/RedisStreamSource.ts [hunk 1]
```diff
@@ -151,35 +153,36 @@ export class RedisStreamSource implements IInputSource, IRequireInitialization,
...
+                    this.config.consumerGroup,
+                    this.config.batchSize
+                );
 
-            // if there are no pending messages return early w/ an empty array
-            if (pendingMessages.length < 1) {
-                return [];
-            }
+                // if there are no pending messages return early w/ an empty array
+                if (pendingMessages.length < 1) {
```
Comment (prachi30) [inline packages/redis/src/RedisStreamSource.ts:169]: nit: `if (!pendingMessages.length)`

Comment (mag415) [reply packages/redis/src/RedisStreamSource.ts:169]: updated

File: packages/redis/src/RedisClient.ts [hunk 2]
```diff
@@ -367,19 +367,19 @@ export class RedisClient implements IRedisClient, IRequireInitialization, IDispo
...
     public async xReadGroup(
         context: SpanContext,
-        streamName: string,
+        streamNames: string[],
         consumerGroup: string,
         consumerName: string,
         count: number,
         block: number,
-        id: string = ">"
+        ids?: string[]
```
Comment (sklose) [inline packages/redis/src/RedisClient.ts:375]: it looks like `ids` isn't actually used anymore. couple of options (not sure what makes most sense here)

1. keep it as `id: string = ">"` and change `streamNames.map(() => ">")` to `streamNames.map(() => id)`
2. streamName + id seem to come in pairs, maybe combine that as a single argument `streams: { name: string; id?: string }[]`;

Comment (mag415) [reply packages/redis/src/RedisClient.ts]: updated to use suggestion #2

**Pull request merged**